### DISCUSSION TITLE
Predicted object callbacks delayed on a clientHost

### DISCUSSION THREAD

**Discussion by Hukha** [FirstGearGames/FishNet#377] [Feature Request]
Basically what the title says. 
When you spawn a Predicted Object in a clientHost, the callbacks (OnStartClient, OnNetworkStart...) for those objects are delayed. This does not happen in normal clients AFAIK.

This has two problems imho:
- First, the clientHost behaves different than the other clients, that is a problem by itself.
- Second, let's say that the clientHost spawns a bullet that then moves if it's the owner, since the callbacks and the IsOwner is delayed, the bullet will spawn and movement will be delayed.

In case someone has this issue, what I did is basically create a layer in-between where my created objects know if they were created "locally" to start the movement even before receiving the network callbacks

**Comment by FirstGearGames** [comment]
What is the feature request here? What is behaving different?

**Reply by Hukha** [reply]
Well, as I said:
A predicted objects spawned in a clientHost has delay in it's callbacks (Also in the setting of IsOwner ofc) while normal clients (not host) have none.
This creates different behaviours between clients and can cause problems like the one described with the bullet.

**Comment by FirstGearGames** [comment]
When the object is spawned as clientHost the server is spawning it because
predicted spawning isn't needed. You should be using OnStartNetwork to
check if owner is local client if you are planning to code for clientHost.
IsOwner will become true when the client end receives the spawn packet.

Perhaps I can look into using predicted spawning as clientHost if the
object is configured to.
